thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -100,65 +102,70 @@ public DetectionTaskRunner() {
     knownAnomalies = getExistingAnomalies();
 
     // Seed request with top-level...
-    TimeSeriesRequest topLevelRequest = new TimeSeriesRequest();
-    topLevelRequest.setCollectionName(anomalyFunctionSpec.getCollection());
+    TimeSeriesRequest request = new TimeSeriesRequest();
+    request.setCollectionName(anomalyFunctionSpec.getCollection());
     List<MetricExpression> metricExpressions = Utils
         .convertToMetricExpressions(metricFunction.getMetricName(),
             anomalyFunctionSpec.getMetricFunction(), anomalyFunctionSpec.getCollection());
-    topLevelRequest.setMetricExpressions(metricExpressions);
-    topLevelRequest.setAggregationTimeGranularity(timeGranularity);
-    topLevelRequest.setStart(windowStart);
-    topLevelRequest.setEnd(windowEnd);
-    topLevelRequest.setEndDateInclusive(false);
+    request.setMetricExpressions(metricExpressions);
+    request.setAggregationTimeGranularity(timeGranularity);
+    request.setEndDateInclusive(false);
     if (StringUtils.isNotBlank(filters)) {
-      topLevelRequest.setFilterSet(ThirdEyeUtils.getFilterSet(filters));
+      request.setFilterSet(ThirdEyeUtils.getFilterSet(filters));
     }
     String exploreDimension = detectionTaskInfo.getGroupByDimension();
     if (StringUtils.isNotBlank(exploreDimension)) {
-      topLevelRequest
+      request
           .setGroupByDimensions(Collections.singletonList(detectionTaskInfo.getGroupByDimension()));
     }
 
-    LOG.info(
-        "Running anomaly detection job with windowStartProp: {}, windowEndProp: {}, metricExpressions: {}, timeGranularity: {}, windowStart: {}, windowEnd: {}",
-        windowStart, windowEnd, metricExpressions, timeGranularity);
+    List<Pair<Long, Long>> startEndTimeRanges =
+        anomalyFunction.getDataRangeIntervals(windowStart.getMillis(), windowEnd.getMillis());
 
-    List<RawAnomalyResultDTO> results = exploreCombination(topLevelRequest);
-    LOG.info("{} anomalies found in total", anomalyCounter);
+    List<TimeSeriesRow> timeSeriesRows = new ArrayList<>();
+
+    for (Pair<Long, Long> startEndInterval : startEndTimeRanges) {
+      DateTime startTime = new DateTime(startEndInterval.getFirst());
+      DateTime endTime = new DateTime(startEndInterval.getSecond());
+      request.setStart(startTime);
+      request.setEnd(endTime);
 
+      LOG.info(
+          "Fetching data with startTime: [{}], endTime: [{}], metricExpressions: [{}], timeGranularity: [{}]",
+          startTime, endTime, metricExpressions, timeGranularity);
+
+      try {
+        LOG.debug("Executing {}", request);
+        TimeSeriesResponse response = timeSeriesHandler.handle(request);
```

Comment:
add a todo here, instead of making multiple calls to pinot, we should enhance timeserieshandler to take multiple ranges and construct appropriate query